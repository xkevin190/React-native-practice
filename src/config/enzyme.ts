import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

/**
 * @tech-debt
 *
 * Update enzyme to support react 18
 */
Enzyme.configure({ adapter: new Adapter() });
