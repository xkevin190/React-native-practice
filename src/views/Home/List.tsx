import React, {useCallback, useState} from 'react';
import {View, StyleSheet, FlatList} from 'react-native';
import Item from '../../components/Item';
import {NavigationProps} from '../../constants/types';

import {item as IItem, ListItems} from '../../state/aplication/type';

interface ListProps {
  ListItems?: ListItems;
  route?: any;
  loadingApp?: boolean;
  setFavorite: (name: string) => void;
}

type renderItemProps = {
  item: IItem;
  index: number;
};

const keyExtractor = (_item: IItem) => _item.name;

const List = (props: ListProps) => {
  const action = (url: string) => {
    props.setFavorite(url);
  };

  const memoizedCallback = useCallback(action, []);

  const renderItem = ({item}: renderItemProps): JSX.Element => {
    return (
      <Item
        gender={item.gender}
        name={item.name}
        homeworld={item.homeworld}
        birth_year={item.birth_year}
        favorite={item.favorite}
        action={memoizedCallback}
      />
    );
  };

  return (
    <FlatList
      data={props.ListItems}
      keyExtractor={keyExtractor}
      renderItem={renderItem}
      removeClippedSubviews={true}
      initialNumToRender={4}
    />
  );
};

export default List;
