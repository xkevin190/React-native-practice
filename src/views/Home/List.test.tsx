import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import List from './List';
import {View} from 'react-native';

describe('List (Unit-Test)', () => {
  let wrapper: ShallowWrapper;

  const favoriteAction = jest.fn();
  const list = [
    {
      gender: 'test',
      name: 'test',
      birth_year: 'test',
      homeworld: 'test',
      favorite: false,
    },
  ];

  beforeEach(() => {
    wrapper = shallow(<List ListItems={list} setFavorite={favoriteAction} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});