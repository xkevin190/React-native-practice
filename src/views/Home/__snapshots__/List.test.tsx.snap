// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List (Unit-Test) should match snapshot 1`] = `
<FlatList
  data={
    Array [
      Object {
        "birth_year": "test",
        "favorite": false,
        "gender": "test",
        "homeworld": "test",
        "name": "test",
      },
    ]
  }
  initialNumToRender={4}
  keyExtractor={[Function]}
  removeClippedSubviews={true}
  renderItem={[Function]}
/>
`;
